src/gamestate/system_state.cpp

```diff
@@ -563,6 +596,10 @@ void state::on_create() {
 		new_elm->set_visible(*this, false);
 		ui_state.root->add_child_to_front(std::move(new_elm));
 	}
+	{
+		auto new_elm = ui::make_element_by_type<ui::map_tooltip>(*this, "tooltip");
```

Comment:
We really should reuse the existing tooltip window, not create a second new window. Having two windows to display tooltips provides no advantage that I can see while opening the door to bugs that simply couldn't exist if there was just one window

Reply:
problem is that im not sure how to avoid it in this case, since a custom update_tooltip() is needed, i suppose i could just make the "map" a element, although that appears hacky

Reply:
That is why we should collaborate on this more. The tooltip doesn't have an internal "update" at all. Rather the render function drives updating the tooltip's content

Reply:
how do you mean? each element has a update_tooltip() function and the render code calls said function no?